cloudevents/formats/cbor-format.md

```diff
@@ -0,0 +1,199 @@
+
+### 1.1. Conformance
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC2119][rfc2119].
+
+## 2. Attributes
+
+This section defines how CloudEvents attributes are mapped to CBOR. This
+specification does not explicitly map each attribute, but provides a generic
+mapping model that applies to all current and future CloudEvents attributes,
+including extensions.
+
+For clarity, extension attributes are encoded using the same rules as
+core attributes. This includes their syntax and placement
+within the CBOR data item. In particular, extensions are placed as top-level CBOR
+key-value pairs of a map. Extensions MUST be encoded as a top-level CBOR key-value pairs.
+
+### 2.1. Base Type System
+
+The core [CloudEvents specification][ce] defines a minimal abstract type system,
+which this mapping leans on.
+
+### 2.2. Type System Mapping
+
+The [CloudEvents type system][ce-types] MUST be mapped to CBOR types as follows,
+with exceptions noted below.
+
+| CloudEvents   | CBOR                                                         |
+| ------------- | ------------------------------------------------------------ |
+| Boolean       | [Simple value][cbor-fpnocont] number `20` for `false` and number `21` for `true` |
+| Integer       | [Major type 0][cbor-major-types] for positive integers and [Major type 1][cbor-major-types] for negative integers                         |
+| String        | [Major type 3][cbor-major-types]                             |
+| Binary        | [Major type 2][cbor-major-types]                             |
+| URI           | [Major type 6][cbor-major-types] [tagged][cbor-tagging] as a URI (tag number `32`)|
+| URI-reference | [Major type 3][cbor-major-types] |
+| Timestamp     | [Major type 6][cbor-major-types] [tagged][cbor-tagging] as a [Standard Date/Time String][cbor-standard-datetime] (tag number `0`)|
+
+Unset attributes MAY be encoded to the [Simple value][cbor-fpnocont] value `22` (`null`). When decoding
+attributes and a `null` value is encountered, it MUST be treated as the
+equivalent of unset or omitted.
+
+Extension specifications MAY define secondary mapping rules for the values of
+attributes they define, but MUST also include the previously defined primary
+mapping.
+
+For instance, the attribute value might be a data structure defined in a
+standard outside of CloudEvents, with a formal CBOR mapping, and there might be
+risk of translation errors or information loss when the original format is not
+preserved.
+
+An extension specification that defines a secondary mapping rule for CBOR, and
+any revision of such a specification, MUST also define explicit mapping rules
+for all other event formats that are part of the CloudEvents core at the time of
+the submission or revision.
+
+If required, the CloudEvents type can be determined by inference using the rules
```

Comment:
s/required/needed/

Reply:
I'm not sure I understand this paragraph. Why is CE `type` called out as special?

Reply:
It is referring to the attribute type, not the event type. 
I copied it from the JSON spec.